### ISSUE TITLE
Bug: Periodic Failure of Consumer and Producer tests.

### ISSUE THREAD

**Issue by peake100** [peake100/rogerRabbit-go#41]
CI Output:

```
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"OQ==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"MA==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"Mg==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"MQ==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"Mw==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"NA==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=NotifyClose ID=default INSTANCE=3 OP_ATTEMPT=-1 TRANSPORT=CHANNEL zARGS="marshaling error: json: unsupported type: chan *amqp.Error" zRESULTS="marshaling error: json: unsupported type: chan *amqp.Error"
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"Ng==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"NQ==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"OA==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
7:41PM DBG  :METHOD_CALL=Publish ID=default INSTANCE=3 OP_ATTEMPT=0 TRANSPORT=CHANNEL zARGS={"Exchange":"","Immediate":false,"Key":"test_queue_producer_publish","Mandatory":false,"Msg":{"AppId":"","Body":"Nw==","ContentEncoding":"","ContentType":"","CorrelationId":"","DeliveryMode":0,"Expiration":"","Headers":null,"MessageId":"","Priority":0,"ReplyTo":"","Timestamp":"0001-01-01T00:00:00Z","Type":"","UserId":""}}
panic: confirmation delivery tag 3 does not line up with order delivery tag 1

goroutine 52 [running]:
github.com/peake100/rogerRabbit-go/roger/amqpproducer.(*Producer).runConfirmations(0xc000296a00)
	/home/vsts/work/1/s/roger/amqpproducer/producer.go:100 +0x51c
github.com/peake100/rogerRabbit-go/roger/amqpproducer.(*Producer).Run.func3(0xc0003bb2e0, 0xc000296a00)
	/home/vsts/work/1/s/roger/amqpproducer/producer.go:295 +0x88
created by github.com/peake100/rogerRabbit-go/roger/amqpproducer.(*Producer).Run
	/home/vsts/work/1/s/roger/amqpproducer/producer.go:293 +0x39b
FAIL	github.com/peake100/rogerRabbit-go/roger/amqpproducer	0.063s
```

**Comment by peake100** [comment]
Closed by #42

**Comment by peake100** [comment]
Just ran into the off-by-one consumer issue. Re-opening

**Comment by peake100** [comment]
closed by #43
